### ISSUE TITLE
<type_traits>: is_layout_compatible does not seem to consider base class members

### ISSUE THREAD

**Issue by BRevzin** [microsoft/STL#1960]
Short example:

```cpp
#include <type_traits>

struct S {
    int* data;
    int len;
};

class C : public S {
};

static_assert(std::is_layout_compatible_v<C, S>);
```

Using MSVC v19.latest on [compiler explorer](https://godbolt.org/z/9v8jYe99v), that assertion fails. But it should succeed: `C` and `S` should be considered to be layout compatible.

**Comment by CaseyCarter** [comment]
Reduces to:
```c++
struct B {
    int x;
};

class D : public B {};

static_assert(__is_layout_compatible(D, B));
```
which indicates this is a bug in the `__is_layout_compatible` compiler intrinsic in MSVC. I've reported this as DevCom-1442026.

**Comment by CaseyCarter** [comment]
DevCom-1442026 has been fixed, almost certainly for VS2022 17.0 Preview 3.
